src/nbl/asset/utils/IShaderCompiler.cpp

```diff
@@ -22,6 +25,73 @@ IShaderCompiler::IShaderCompiler(core::smart_refctd_ptr<system::ISystem>&& syste
     m_defaultIncludeFinder = core::make_smart_refctd_ptr<CIncludeFinder>(core::smart_refctd_ptr(m_system));
 }
 
+static void* SzAlloc(ISzAllocPtr p, size_t size) { p = p; return _NBL_ALIGNED_MALLOC(size, _NBL_SIMD_ALIGNMENT); }
+static void SzFree(ISzAllocPtr p, void* address) { p = p; _NBL_ALIGNED_FREE(address); }
+
+inline core::smart_refctd_ptr<ICPUShader> nbl::asset::IShaderCompiler::compileToSPIRV(const std::string_view code, const SCompilerOptions& options) const
+{
+    CCache::SEntry entry;
+    std::vector<CCache::SEntry::SPreprocessingDependency> dependencies;
+    if (options.readCache || options.writeCache)
+        entry = std::move(CCache::SEntry(code, options));
+
+    if (options.readCache)
+    {
+        auto found = options.readCache->find_impl(entry, options.preprocessorOptions.includeFinder);
+        if (found != options.readCache->m_container.end())
+        {
+            if (options.writeCache)
+            {
+                CCache::SEntry writeEntry = *found;
+                options.writeCache->insert(std::move(writeEntry));
+            }
+            return found->decodeShader();
+        }
+    }
+
+    auto retVal = compileToSPIRV_impl(code, options, options.writeCache ? &dependencies : nullptr);
+    // compute the SPIR-V shader content hash
+    {
+        auto backingBuffer = retVal->getContent();
+        const_cast<ICPUBuffer*>(backingBuffer)->setContentHash(backingBuffer->computeContentHash());
+    }
+
+    std::cout << "writeCache: " << options.writeCache << "\n";
```

Comment:
we have our own logger interface for a reason

Reply:
debug printing catched